extensions/common/vault/vault-hashicorp/src/main/java/org/eclipse/edc/vault/hashicorp/HashicorpVaultClient.java

```diff
@@ -24,76 +25,192 @@
     private static final int HTTP_CODE_404 = 404;
-    @NotNull
-    private final HashicorpVaultClientConfig hashicorpVaultConfig;
+    private static final TypeReference<Map<String, Object>> MAP_TYPE_REFERENCE = new TypeReference<>() {
+    };
+
     @NotNull
     private final EdcHttpClient httpClient;
     @NotNull
+    private final Headers headers;
+    @NotNull
     private final ObjectMapper objectMapper;
+    @NotNull
+    private final Monitor monitor;
+    @NotNull
+    private final HashicorpVaultConfigValues configValues;
+    @NotNull
+    private final HttpUrl healthCheckUrl;
 
-    HashicorpVaultClient(@NotNull HashicorpVaultClientConfig hashicorpVaultConfig, @NotNull EdcHttpClient httpClient,
-                         @NotNull ObjectMapper objectMapper) {
-        this.hashicorpVaultConfig = hashicorpVaultConfig;
+    HashicorpVaultClient(@NotNull EdcHttpClient httpClient,
+                         @NotNull ObjectMapper objectMapper,
+                         @NotNull Monitor monitor,
+                         @NotNull HashicorpVaultConfigValues configValues) {
         this.httpClient = httpClient;
         this.objectMapper = objectMapper;
+        this.monitor = monitor;
+        this.configValues = configValues;
+        this.headers = getHeaders();
+        this.healthCheckUrl = getHealthCheckUrl();
+    }
+
+    public Result<Void> doHealthCheck() {
+        var request = httpGet(healthCheckUrl);
+
+        try (var response = httpClient.execute(request)) {
+            if (response.isSuccessful()) {
+                return Result.success();
+            }
+
+            var code = response.code();
+            var errMsg = switch (code) {
+                case 429 -> "Vault is in standby";
+                case 472 -> "Vault is in recovery mode";
+                case 473 -> "Vault is in performance standby";
+                case 501 -> "Vault is not initialized";
+                case 503 -> "Vault is sealed";
+                default -> "Vault returned unspecified code %d".formatted(code);
+            };
+            var body = response.body();
+            if (body == null) {
+                return Result.failure("Healthcheck returned empty response body");
+            }
+            return Result.failure("Vault is not available. Reason: %s, additional information: %s".formatted(errMsg, body.string()));
+        } catch (IOException e) {
+            var errMsg = "Failed to perform healthcheck with reason: %s".formatted(e.getMessage());
+            monitor.warning(errMsg, e);
+            return Result.failure(errMsg);
+        }
+    }
+
+    /**
+     * Attempts to look up the Vault token defined in the configurations.
+     * <p>
+     * Will retry in some error cases.
+     *
+     * @return the result of the token lookup operation
+     */
+    public Result<Map<String, Object>> lookUpToken() {
```

Comment:
I'm not an huge fan of these "raw" return objects, because then the caller need to know how to get the data out of it. As the only usage of this response body extracts the `isRenevable` boolean, I would make this method return a `Result<Boolean>` so the caller won't need to deserialize further